lib/Reference/TableReferenceProvider.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace OCA\Tables\Reference;
+
+use Exception;
+use OC\Collaboration\Reference\LinkReferenceProvider;
+use OCA\Tables\Service\TableService;
+use OCP\Collaboration\Reference\IReferenceProvider;
+use OCP\Collaboration\Reference\Reference;
+use OC\Collaboration\Reference\ReferenceManager;
+use OCA\Tables\AppInfo\Application;
+use OCP\Collaboration\Reference\IReference;
+use OCP\IURLGenerator;
+use Throwable;
+
+class TableReferenceProvider implements IReferenceProvider {
+	private const RICH_OBJECT_TYPE = Application::APP_ID . '_table';
+
+	private ?string $userId;
+	private ReferenceManager $referenceManager;
+	private IURLGenerator $urlGenerator;
+	private LinkReferenceProvider $linkReferenceProvider;
+	private TableService $tableService;
+
+	public function __construct(IURLGenerator $urlGenerator,
+								TableService $tableService,
+								ReferenceManager $referenceManager,
+								LinkReferenceProvider $linkReferenceProvider,
+								?string $userId) {
+		$this->userId = $userId;
+		$this->referenceManager = $referenceManager;
+		$this->urlGenerator = $urlGenerator;
+		$this->linkReferenceProvider = $linkReferenceProvider;
+		$this->tableService = $tableService;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function matchReference(string $referenceText): bool {
+		if ($this->userId === null) {
+			return false;
+		}
+		$start = $this->urlGenerator->getAbsoluteURL('/apps/' . Application::APP_ID);
+		$startIndex = $this->urlGenerator->getAbsoluteURL('/index.php/apps/' . Application::APP_ID);
+
+		// link example:
+		// https://nextcloud.local/apps/tables/#/table/3
+		$noIndexMatch = preg_match('/^' . preg_quote($start, '/') . '\/#\/table\/\d+$/i', $referenceText) === 1;
+		$indexMatch = preg_match('/^' . preg_quote($startIndex, '/') . '\/#\/table\/\d+$/i', $referenceText) === 1;
+
+		return $noIndexMatch || $indexMatch;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function resolveReference(string $referenceText): ?IReference {
+		if ($this->matchReference($referenceText)) {
+			$tableId = $this->getTableIdFromLink($referenceText);
+			if ($tableId === null) {
+				// fallback to opengraph if it matches but somehow we can't resolve
+				return $this->linkReferenceProvider->resolveReference($referenceText);
+			}
+			try {
+				$table = $this->tableService->find($tableId, $this->userId)->jsonSerialize();
```

Comment:
Any reason to serialise here here already instead of using the entity object further down?

Reply:
`OCA\Tables\Db\Table` does not declare the getters so I didn't use them and went for accessing the info after having serialized. No special reason. Also, I figured this would be rewritten anyway :grin:. This PR's goal is just to provide a basic implementation to be polished then.